src/db/delete.rs

```diff
@@ -13,26 +12,27 @@ enum CrateDeletionError {
     MissingCrate(String),
 }
 
-pub fn delete_crate(conn: &Connection, name: &str) -> Result<(), Error> {
+pub fn delete_crate(conn: &Connection, storage: &Storage, name: &str) -> Result<(), Error> {
     let crate_id = get_id(conn, name)?;
     delete_crate_from_database(conn, name, crate_id)?;
 
-    if let Some(client) = s3_client() {
-        for prefix in STORAGE_PATHS_TO_DELETE {
-            delete_prefix_from_s3(&client, &format!("{}/{}/", prefix, name))?;
-        }
+    for prefix in STORAGE_PATHS_TO_DELETE {
+        storage.delete_prefix(&format!("{}/{}/", prefix, name))?;
     }
 
     Ok(())
 }
 
-pub fn delete_version(conn: &Connection, name: &str, version: &str) -> Result<(), Error> {
+pub fn delete_version(
+    conn: &Connection,
+    storage: &Storage,
+    name: &str,
+    version: &str,
+) -> Result<(), Error> {
```

Comment:
Maybe switch away from the standalone function since you've already moved the meat of it to the associated function on `Storage`